src/analysis/inner_product_space/pi_L2.lean

```diff
@@ -150,54 +150,198 @@ begin
   simp [e₂, direct_sum.submodule_coe, direct_sum.to_module, dfinsupp.sum_add_hom_apply]
 end
 
-/-- An orthonormal basis on a fintype `ι` for an inner product space induces an isometry with
+variables (ι) (𝕜) (E)
+
+/-- The vector given in euclidean space by being `1 : 𝕜` at coordinate `i : ι` and `0 : 𝕜` at
+all other coordinates. -/
+def euclidean_space.single {𝕜 : Type*} {ι : Type*} [fintype ι] [is_R_or_C 𝕜] (i : ι) (a : 𝕜) :
+ euclidean_space 𝕜 ι :=
+  set.indicator {i} (λ j, a)
+
+
+@[simp] theorem euclidean_space.single_apply {𝕜 : Type*} {ι : Type*} [fintype ι] [decidable_eq ι]
+  [is_R_or_C 𝕜] (i : ι) (a : 𝕜) (j : ι) :
+  (euclidean_space.single i a) j = ite (j = i) a 0 :=
+begin
+  rw [euclidean_space.single, set.indicator],
+  simp only [set.mem_singleton_iff],
+end
+
+lemma euclidean_space.inner_single_left (i : ι) (a : 𝕜) (v : euclidean_space 𝕜 ι) :
+  ⟪ euclidean_space.single i (a : 𝕜), v ⟫ = star_ring_end 𝕜 a * (v i) :=
```

Comment:
Couple of style points:
```suggestion
  ⟪euclidean_space.single i (a : 𝕜), v⟫ = conj a * v i :=
```
(note the spacing with `⟪ ⟫` and the more readable notation `conj` for "star_ring_end").  Can you also do the same with the next lemma, and also shorten the proof of the next lemma by the same techniques used for this one?